### ISSUE TITLE
Add ability to track messages being displayed to users

### ISSUE THREAD

**Issue by rchiodo** [microsoft/pyright#5249]
We're about to localize all of our error messages (@bschnurr is currently working on this for pylance and pyright). 

We'd like to add telemetry to track what the most common error messages are so we can provide better documentation.

Is there some way to track what errors are being reported? 

I think we might be able to this in a diagnostic sink:
https://github.com/microsoft/pyrx/main/packages/pyright/packages/pyright-internal/src/common/diagnosticSink.ts#L78

but the localization of the strings makes identifying the message hard.

**Comment by erictraut** [comment]
> We're about to localize all of our error messages

That's great to hear!

One potential solution is to hook in to `getRawString`. This will give you unique string identifiers and counts. There are some strings that could be associated with suppressed diagnostics (e.g. because someone used a `# type: ignore` or because the diagnostic is associated with a file outside of a tracked or open file), but these are probably a small enough percentage that they wouldn't mess up the stats in an appreciable way.

**Comment by erictraut** [comment]
We might want to do a full pass over all of the error messages for consistency and readability. I've tried to keep them pretty consistent, but I admit that there's room for improvement here.

The telemetry could help us focus on improving the readability of the most common errors.

**Comment by erictraut** [comment]
@rchiodo, I presume that you're signing up to implement this? This might be better tracked in the pylance issue tracker since it's related to telemetry, and pyright has no telemetry.

**Comment by rchiodo** [comment]
Yes, I'm probably going to be the one to implement this. Wanted to get your ideas here first though. I guess that doesn't matter as you watch pylance-release as well. I'll transfer it.
